gateway/src/shard/impl.rs

```diff
@@ -697,8 +703,10 @@ impl Shard {
             kind: SendErrorType::SessionInactive,
         })?;
 
-        if let Some(ratelimiter) = session.ratelimit.get() {
-            ratelimiter.acquire_one().await;
+        if let Some(maybe_ratelimiter) = session.ratelimit.get() {
+            if let Some(ratelimiter) = maybe_ratelimiter {
+                ratelimiter.acquire_one().await;
+            }
         } else {
             return Err(SendError {
                 kind: SendErrorType::HeartbeaterNotStarted,
```

Comment:
I believe we need to put the `else` inside this branch:
```rust
if let Some(maybe_ratelimiter) = session.ratelimit.get() {
```

Reply:
No, the logic is fine this way. I've added a comment in the code to clarify it.